REACH-69 DynamoWebServerTests: File open/save/close

marimano opened the pull request:
[Link to the task](http://adsk-oss.myjetbrains.com/youtrack/issue/REACH-69)

marimano commented on test/core/commands/codeBlockExpectedResult.txt:
```diff
@@ -1 +1 @@
-{"Code":"A = B;\nC=1;", "LineIndices": [0, 1],"InPorts": ["B"], "OutPorts": ["A","C"], "Data": "null"}
\ No newline at end of file
+{"Code":"A = B;\nC=1;", "LineIndices": [0, 1],"InPorts": ["B"], "OutPorts": ["A","C"]}
```
To pass `CanSendCodeBlockNodeUpdate()` test case correctly it requires changes from [this PR](https://github.com/Benglin/Dynamo/pull/256) where `Data` is not being included anymore

Benglin replied:
This is the right result of the test, so it looks good. Let it fail for now and after another PR gets in, it should be passing.

marimano commented:
Hi @Benglin, please take a look at these tests

Benglin commented on test/Server/SocketTest/SaveUploadTests.cs:
```diff
@@ -0,0 +1,379 @@
+
+        #endregion
+
+        #region Private helping methods
```
Private *helper* methods ;)

Benglin commented on test/Server/SocketTest/SaveUploadTests.cs:
```diff
@@ -0,0 +1,379 @@
+        [SetUp]
+        public override void Init()
+        {
+            base.Init();
```
This looks good, I just need confirmation that the base `DynamoModel.Start` gets called exactly once per test case (because you also have `[SetUp]` attribute, so I'm not sure how the initialization process goes).

Benglin commented on test/Server/SocketTest/SaveUploadTests.cs:
```diff
@@ -0,0 +1,379 @@
+            bool proxyResponseTookPlace = false;
+
+            messageHandler.ResultReady += (s, args) =>
+            {
```
Since `DynamoModel.StartInTestMode` is set to `true` for all `DynamoViewModelUnitTest` derived class, this will be called synchronously (i.e. no background thread that takes over execution and returns at a later time). So that is good. Just need to confirm that this callback is actually invoked (because without it invoked the tests will pass)?

marimano replied:
I run all these tests and they all passed successfully

Benglin commented on test/Server/SocketTest/SaveUploadTests.cs:
```diff
@@ -0,0 +1,379 @@
+                }
+                else
+                {
+                    Assert.IsTrue(false, WebServerErrorMessages.WrongResponse);
```
Perhaps `Assert.Fail` is more readable?

Benglin commented:
Thanks @marimano, these look good. I didn't really examine the contents of those command files to make sure the workflow is meaningful, and that you get the right results back. If you could give me those confirmations above, then this is good to merge.

marimano commented:
Hi @Benglin, I've updated the PR a bit.
 
As to command files I had added many different nodes into them on purpose because most often I detects some issues when I test a functionality with many nodes in a workspace whereas everything works fine with few nodes in a workspace. So those big files are for making sure everything works as it should do :)

I run all these tests and they all passed successfully. So I consider it can be merged in

Benglin commented:
Thanks for the confirmation, @marimano. LGTM! (Sorry again for the delay)
